src/bsoncxx/types/value.cpp

```diff
@@ -30,8 +30,6 @@ namespace types {
 #define BSONCXX_ENUM(name, val)                                                                \
     value::value(b_##name value) noexcept : _type(static_cast<bsoncxx::type>(val)),            \
                                             _b_##name(std::move(value)) {                      \
-        static_assert(std::is_nothrow_copy_constructible<b_##name>::value, "Copy may throw");  \
```

Comment:
I've fought the good fight so far on avoiding use of macros, but I'm inclined to make the inclusion of the static asserts conditional on the source of polyfills.